[AY1920S1-CS2113-F14-4] SpinBox

rssujay opened the pull request:
@TCK1997
@ttippinyu
@lackmoo

rssujay committed:
Port over Phase 1 codebase

rssujay committed:
Merge pull request #1 from rssujay/original

Port over Phase 1 codebase

rssujay committed:
Add post-port consistency changes

TCK1997 committed:
Merge pull request #3 from rssujay/code-port-consistency

Add post-port consistency changes

rssujay committed:
Add Travis CI

ttippinyu committed:
Merge pull request #4 from rssujay/travis-ci

Add Travis CI

rssujay committed:
Update build.gradle product version (#5)

rssujay committed:
Add checkStyle reminder for 2113T styling (#6)

rssujay committed:
Retitle jar filename (#7)

rssujay committed:
Fix missing new directory during setup (#8)

TCK1997 committed:
Add library natty (#9)

* Added library natty

Added library natty in build.gradle.
Changed some classes to accomodate changes.
DateTime now wraps java.util.Date

* fixed checkstyle error

* Change date format in error message

rssujay committed:
Update storage unit tests and maintenance (#11)

lackmoo committed:
Add FixedDuration Tasks (#12)

* Add Fixed Duration Tasks Extension

* Add JavaDoc comment for Fixed.java

* Update FixedTest with correct expected output

* Edit typo in AddCommand and change test unit names

ttippinyu committed:
Fix bug when done or delete input has no index (#14)

* fix bug when done has no index

* Fix done and delete no index inputted bug

rssujay committed:
Add B-Reminders and fix bugs (#16)

* Fix storage and empty string bug

* Rename methods, make Tasks abstract

* Add Reminders for tasks

* Troubleshoot unit test

* Fix unit test relying on timezone

lackmoo committed:
Add B-DoAfterTasks (#15)

ttippinyu committed:
Implement B-DoWithinPeriodTasks (#19)

* add do-within-period feature

* checkstyle update

* checkstyle update

* fixed output message in exception handling in AddCommand

* fix WithinTest dates check

ttippinyu committed:
Implement B-ViewSchedules  (#18)

* temp for view-schedule

* Added view-schedule feature

* Checkstyle update

* Checkstyle update

* change compareTo to compareEquals to get rid of instanceof

* update compareEquals

* update ViewScheduleTest

rssujay committed:
Implement B-RecurringTasks (#20)

* Implement recurring tasks

* Add unit tests

* Reorganize test cases

Commit:
Add AboutUs.adoc and individual images

lackmoo committed:
Add AboutUs.adoc and individual images (#22)

ttippinyu committed:
Implement B-TentativeScheduling (#21)

* add tentative scheduling

* took out system.out debugging

* save task after SetCommand

TCK1997 committed:
Implement B-Snooze (#10)

* Added library natty

Added library natty in build.gradle.
Changed some classes to accomodate changes.
DateTime now wraps java.util.Date

* fixed checkstyle error

* Change date format in error message

* Add Snooze Command

Add SnoozeCommand class.
Modified Parser class.
Add setBy method into Deadline class.
Add setStart and setEnd method into Event class.

* Add JUnit Test for B-Snooze

* Edit SnoozeTest.java

Commit:
Merge remote-tracking branch 'upstream/master'

Commit:
Update README.md

Commit:
Edit README.md

Commit:
Update README.md

Commit:
Update README.md

Commit:
Add README.adoc

Commit:
Update README.md

Commit:
Update README.adoc

lackmoo committed:
Update README.md (#32)

* Add AboutUs.adoc and individual images

* Update README.md

* Edit README.md

* Update README.md

* Update README.md

* Add README.adoc

* Update README.md

* Update README.adoc

Commit:
Merge remote-tracking branch 'upstream/master'

TCK1997 committed:
Implement B-DetectAnomalies (#33)

* Added library natty

Added library natty in build.gradle.
Changed some classes to accomodate changes.
DateTime now wraps java.util.Date

* fixed checkstyle error

* Change date format in error message

* Check for conflicting time interval

Implement isOverlapping function in Task.
Check if event will conflict with other task right before adding.

* Add JUnit test for B-DetectAnomalies

* Fix minor checkstyle issues in test

TCK1997 committed:
Implement B-FindFreeTimes (#34)

* Added library natty

Added library natty in build.gradle.
Changed some classes to accomodate changes.
DateTime now wraps java.util.Date

* fixed checkstyle error

* Change date format in error message

* Fix Exception Message of Do-After

* Add Find Free Time functionality

Commit:
Merge remote-tracking branch 'upstream/master'

ttippinyu committed:
Add file tracking feature (#62)

Commit:
Merge remote-tracking branch 'upstream/master'

rssujay committed:
Refactor tasks (#63)

* Methods common to multiple tasks added to abstract class Tasks

* Further refactoring should be done post-architectural decisions

lackmoo committed:
Implement delete multiple tasks feature (#64)

* Implement Delete Multiple Command and change Done Command implementation

Commit:
Merge remote-tracking branch 'upstream/master'

Commit:
Add Ui.png

Commit:
Edit README

Commit:
Resize individual images

Commit:
Resize individual images

lackmoo committed:
Include Ui mock-up and resize individual images (#65)

* Add AboutUs.adoc and individual images

* Update README.adoc

* Add Ui.png

* Resize individual images

TCK1997 committed:
Added JUnit tests for DateTime and Storage (#66)

* fixed checkstyle error

* Change date format in error message

* Add JUnit test for DateTime

Edit DateTimeTest for some JUnit test
Edit DateTime JavaDoc comment

* Add StorageTest

* Fix code style in StorageTest.java

rssujay committed:
Implement Item & GradedComponent classes (#68)

* Create Item class

* Add GradedComponent subclass

Commit:
Merge remote-tracking branch 'upstream/master'

Commit:
Implement multiple feature to delete multiple or done multiple tasks

ttippinyu committed:
Merge pull request #69 from lackmoo/Multiple

Implement delete and done multiple command feature

ttippinyu committed:
Implement List Classes (#70)

* List classes

TCK1997 committed:
Refactor all Date parsing into DateTime (#71)

Encapsulate natty Parser inside of DateTime to change all parsing of
Date Strings to be done inside of DateTime object.

ttippinyu committed:
Add a basic module class (#73)

* basic module class

ttippinyu committed:
Refactor to SpinBox (#75)

Product morph related refactoring

lackmoo committed:
Implement CommandBuilder Feature within Parser (#76)

TCK1997 committed:
Edit items/task according to architecture (#81)

Delete unnecessary task class.
Add Schedulable and NonSchedulable abstract class.
Inherit all tasks under Schedulable and NonSchedulable.
Add Exam, Lab, Lecture and Tutorial class.
Edit other misc class to accomodate changes.

lackmoo committed:
Update command builder (#83)

* Implement CommandBuilder Feature within Parser

* Update Command Builder

* update javadoc

* checkstyle update

ttippinyu committed:
Update module class (#85)

* List classes

ttippinyu committed:
Move Parser into SpinBox (#86)

* parser into spinbox, add view command

rssujay committed:
Implement hierarchy-based storage (#87)

* Modify lists to contain Storage

* Implement hierarchical storage

* Fix issues and implement unit tests

* Fix paths

rssujay committed:
Implement notepad feature for modules (#88)

* Add notepad and rework exceptions

* Reorganize structure

* Add unit tests for notepad

* Add test module container to unit tests

rssujay committed:
Add minimal support for module addition (#89)

* Add minimal support for module addition

* Fix unit test potential conflicts

ttippinyu committed:
Call sort method in add, replace, and mark (#91)

* sort after add, replace, and mark list

ttippinyu committed:
Enhance AddCommand for Files (#90)

* full and partial add file command

ttippinyu committed:
View lists when on tabs page (#92)

* output list for tabs

ttippinyu committed:
Enhance view command (#93)

* enhance viewCommand

ttippinyu committed:
Remove command for files (#94)

* remove file

* check update

rssujay committed:
Notepad commands (#95)

* Add view and add support for notes

* Implement remove command for notes

lackmoo committed:
Add Task Commands into Add Command (#97)

* Add Task Commands into Add Command

ttippinyu committed:
Mark file command (#96)

* mark file

rssujay committed:
Update .travis.yml (#101)

TCK1997 committed:
Implement Calendar (#98)

* Add Calendar

Modify DateTime, TaskList to accomodate Calendar.
Add CalendarSelectorException.

lackmoo committed:
Implement remove-multiple command, mark command for tasks and fix bugs (#103)

* Implement Remove-multiple command and fix bugs

* Implement mark task command and revise bug fixes

rssujay committed:
Fix duplication bug (#105)

rssujay committed:
Update version and deprecated api call (#104)

lackmoo committed:
Fix typos (#106)

ttippinyu committed:
Update parser (#111)

* update parser

ttippinyu committed:
Update parser and commands (#115)

* List classes

* checkstyle and test update

* deconflict

* deconflict

* mark file

* resolved changes

* update parser

* change error message

* remove print

* update parser and commands

* string literals + case-insensitive

* update string literal

lackmoo committed:
Implement Help Command (#116)

TCK1997 committed:
Add initial GUI layout and navigability (#118)

Added initial GUI layout and navigability
Add TaskBox with TaskBox.java
Edit MainWindow to update individual windows.

langyanbin0314 commented on src/main/java/spinbox/Parser.java:
```diff
@@ -0,0 +1,151 @@
+                tempPageTrace.removeLast();
+                if (tempPageTrace.size() != 0) {
+                    pageData = pageData.concat(" " + tempPageTrace.getLast());
+                }
```
consider using enum instead of string

langyanbin0314 commented on src/main/java/spinbox/Parser.java:
```diff
@@ -0,0 +1,151 @@
+        //            }
+        //        } catch (NumberFormatException e) {
+        //            throw new InputException("Please enter an integer for index");
+        //        }
```
delete unwanted code, do not comment out

langyanbin0314 commented on src/main/java/spinbox/commands/AddCommand.java:
```diff
@@ -0,0 +1,224 @@
+import java.util.List;
+
+public class AddCommand extends Command {
+    private static final String MODULE_ADDED = "The following module has been added to SpinBox: ";
```
Good Job on extracting strings into variables. Keep it up!

langyanbin0314 commented on src/main/java/spinbox/commands/AddCommand.java:
```diff
@@ -0,0 +1,224 @@
+        DateTime start;
+        DateTime end;
+        switch (type) {
+        case "file":
```
consider using enum

langyanbin0314 reviewed (commented):
Good job on removing magic strings in your code.
Should use enum for switch cases

TCK1997 committed:
Add more GUI functionality (#119)

Edit MainWindow to update individual windows.

* Add more GUI Functionality

ttippinyu committed:
Change Mark to Update (#121)


* mark to update

* update help and sorting

lackmoo committed:
Implement different event type tasks and fix bug (#124)

Closes #117

ttippinyu committed:
Implement Find command (#123)

* implement find command

Closes #122

ttippinyu committed:
Make module code case-insensitive (#127)

* List classes

* checkstyle and test update

* deconflict

* deconflict

* moduleCode case insensitive

lackmoo committed:
Implement UpdateMultiple Command (#128)

rssujay committed:
Add functional GUI (#129)

* Set up virtual tabs for modules

* Add files and tasks view on GUI

lackmoo committed:
Implement SetName Command (#130)

* Implement SetName Command

rssujay committed:
Implement interface for containers w/ Storage (#136)

* Add StorageContainer interface

* Let Notepad implement StorageContainer

lackmoo committed:
Implement set date command (#137)

* Edit Commands of Multiple Command

* Implement Set Date command

rssujay committed:
Restructure tests (#138)

ttippinyu committed:
Reorganize commands (#139)

* made command abstract

rssujay committed:
Implement export to text framework (#140)

* Add generic export framework

* Update help pages

ttippinyu committed:
Write integration test for parser & view command (#141)

* parser viewCommand test

rssujay committed:
Implement graded component commands and GUI representation (#142)

* Incorporate GradedComponents into AddCommand

* Add remove and update grades commands

* Add scoring for gradedComponents

* Implement score command and GUI representation

lackmoo committed:
Update Ui.png (#143)

lackmoo committed:
Fix Task Update in Gui (#176)

* Fix Task Update in Gui

rssujay committed:
Add line numbers and header to notes (#177)

rssujay committed:
Attempt add coverall integration (#178)

* Attempt add coverall integration

rssujay committed:
Add storable interface and cleanup (#179)

* Add storable interface and cleanup

* Add javadocs for gradedComponent method

rssujay committed:
Add more test coverage to notepad (#180)

lackmoo committed:
Implement cross-platform support for JAR files (#181)

TCK1997 committed:
Add Calendar view for month (#182)

* Add Calendar view for month

lackmoo committed:
Update gui tasks display (#183)

* Add index to task display

* Update Gui Tasks Display to show index and done status

lackmoo committed:
Command integration test (#185)

* Add Task and Commands Integration Test

* Add Integration Tests for Set-Date and Set-Name

lackmoo committed:
Modify Command integration tests (#186)

* Add Task and Commands Integration Test

ttippinyu committed:
Add File list integration test (#188)

* file list test

ttippinyu committed:
Task list integration test (#190)

* List classes

* checkstyle and test update

* deconflict

* deconflict

* file list test

* task list test

* renaming

ttippinyu committed:
File list integration test (#192)

* List classes

* checkstyle and test update

* deconflict

* deconflict

* file list test

* renaming

lackmoo committed:
Modify Parser Test and Create Task Tests (#191)

* Modify Parser Test and Create Task Unit Tests

lackmoo committed:
Implement Remove Modules Command (#194)

* Implement Remove Modules Command

lackmoo committed:
Update GUI to include index for files and grades view (#195)

* Update GUI to Include Index for Files and Grades tab

TCK1997 committed:
Fix GUI bugs and update tests (#196)

* Fix bug and GUI enhancements

Fix bug regarding calendar refresh
Fix bug regarding view / module module code
Enhance tab to be wider with background color

* Update ParserView Integration Test

lackmoo committed:
Change Update Command Format (#200)

lackmoo committed:
Update remove modules command  (#201)

* Implement Remove Modules Command

rssujay committed:
Implement export all deadlines (#202)

* Implement export all deadlines

* Update help pages

rssujay committed:
Fix popup blocking input box (#203)

TCK1997 committed:
Add Exception for Schedulable tasks (#204)

Add checkValidEndDate method for Schedulable.
Update Schedulable children constructor to check if end date is valid.
Update SetDateCommand to ensure SetDate do not remove task if command
fails.
Update tests with invalid end dates.

TCK1997 committed:
Add module code to calendar and main (#206)

rssujay committed:
Add command to populate sample data (#205)

* Add command to populate SpinBox with data

* Add functionality for populate command

* Extract out magic string

TCK1997 committed:
Add ExamView to Main (#208)

Add ExamView to MainView
Filter out Exams from Urgent Task
Resize TaskBox
Add header to MainView

lackmoo committed:
Implement Cross Platform Support for SpinBox (#207)

TCK1997 committed:
Add Exception Handling for Schedulable tasks (#209)

Add Exception Handling for Schedulable Tasks.
Schedulable Tasks are now unable to add something with start date in the
past.

* Fix tests where date is earlier than now

rssujay committed:
Add assertations and fortify GUI (#210)

rssujay committed:
Propagate corrupted data GUI warning (#211)

* Add assertations and fortify GUI

* Propagate corrupted data GUI warning

lackmoo committed:
Edit Test Module Name and Add Integration Tests (#212)

rssujay committed:
Incorporate logging (#213)

* Incorporate logging

* Modify error message to reference logfile

rssujay committed:
Extract out filehandlers (#215)

* Extract out filehandlers

* Extract out methods

lackmoo committed:
Refactor Parser (#214)

* Refactor Parser
* Fix Assertion Statements

TCK1997 committed:
Add Exception handling to DateTime (#216)

Check date and time string input to DateTime object to ensure correct
input. throws DateFormatException in the event of wrong input by user.

ttippinyu committed:
Refactor List Classes (#218)

* refactored lists

lackmoo committed:
Add Integration Tests for Help Command and Help List (#217)

* Edit Test Module Name and Add Integration Tests

* Create Integration Tests for Help Command and HelpList

lackmoo committed:
Update GUI for Main Page (#219)

ttippinyu committed:
Refactor view command (#220)

* refactor view command

* reduce complex bool

TCK1997 committed:
Improve Calendar GUI and view command (#221)

* Improve Calendar GUI and view command

Add selection function for calendar view command.
Add scrolling funtionality for large amounts of task in a single day for
Calendar GUI.
GUI slight resizing.

* Update header and help command

ttippinyu committed:
Rewrite javaDocs (#222)

* slight refactoring

ttippinyu committed:
Extend find command to graded components (#223)

* find for grades

ttippinyu committed:
Fix Find command whitespace bug (#227)

* find whitespace bug and log

ttippinyu committed:
Refactor Find Command (#228)

* find duplication code

* remove unused import

rssujay committed:
Update release filename (#229)

lackmoo committed:
Refactor Test Codes and Set Date (#230)

* Refactor Parser

* More Refactoring on Parser

* Fix Checkstyle

* Fix Assertion Statements

* Refactor Test Codes

* Refactor SetDateCommand

* Fix set-name command

* Fix set-name command

* Fix set-name command bugs

ttippinyu committed:
Typo (#231)

* List classes

* checkstyle and test update

* deconflict

* deconflict

* typo

ttippinyu committed:
View calendar refactor (#232)

* List classes

* checkstyle and test update

* deconflict

* deconflict

* refactor view

lackmoo committed:
Update Documentation (#233)

lackmoo committed:
Update Documentation (#234)

* Update Documentation

* Update Documentation

* DG

rssujay committed:
Update README.adoc

lackmoo committed:
Update Documentation (#235)

* Update Documentation

* Update Documentation

* PPP

lackmoo committed:
Final Update (#236)
